File: Arrays/Count-Subarrays-with-Given-XOR/Python/Count-Subarrays-with-Given-XOR-Brute Force.py
class Solution:
    def subarrayXor(self, arr: list[int], m: int) -> int:
        n = len(arr)
        count = 0

        for i in range(n):
            xr = 0

            for j in range(i, n):
                xr ^= arr[j]

                if xr == m:
                    count += 1

        return count

File: Arrays/Count-Subarrays-with-Given-XOR/JavaScript/Count-Subarrays-with-Given-XOR-Brute Force.js
var subarrayXor = function(arr, m) {
    const n = arr.length;
    let count = 0;

    for (let i = 0; i < n; i++) {
        let xr = 0;

        for (let j = i; j < n; j++) {
            xr ^= arr[j];

            if (xr === m) {
                count++;
            }
        }
    }

    return count;
};